src/main/java/org/itsallcode/openfasttrace/report/ReportService.java

```diff
@@ -30,35 +30,57 @@
 
 import org.itsallcode.openfasttrace.core.Newline;
 import org.itsallcode.openfasttrace.core.Trace;
+import org.itsallcode.openfasttrace.report.html.HtmlReport;
+import org.itsallcode.openfasttrace.report.plaintext.PlainTextReport;
 
 public class ReportService
 {
-    public void reportTraceToPath(final Trace trace, final Path outputPath,
+    public void reportTraceToPath(final Trace trace, final Path outputPath, final String format,
             final ReportVerbosity verbosity, final Newline newline)
     {
         try (OutputStream outputStream = Files.newOutputStream(outputPath))
         {
-            reportTraceToStream(trace, verbosity, newline, outputStream);
+            reportTraceToStream(trace, format, verbosity, newline, outputStream);
         }
         catch (final IOException e)
         {
-            throw new ReportException("Error generating report to output path " + outputPath, e);
+            throw new ReportException("Error generating stream to output path " + outputPath, e);
         }
     }
 
-    public void reportTraceToStdOut(final Trace trace, final ReportVerbosity verbosity,
-            final Newline newline)
+    public void reportTraceToStdOut(final Trace trace, final String format,
+            final ReportVerbosity verbosity, final Newline newline)
     {
 
-        reportTraceToStream(trace, verbosity, newline, System.out);
+        reportTraceToStream(trace, format, verbosity, newline, System.out);
     }
 
-    private void reportTraceToStream(final Trace trace, final ReportVerbosity verbosity,
-            final Newline newline, final OutputStream outputStream)
+    private void reportTraceToStream(final Trace trace, final String format,
+            final ReportVerbosity verbosity, final Newline newline, final OutputStream outputStream)
     {
         final OutputStream bufferedOutputStream = new BufferedOutputStream(outputStream);
-        new PlainTextReport(trace, newline).renderToStreamWithVerbosityLevel(bufferedOutputStream,
-                verbosity);
+        switch (ReportFormat.parse(format))
+        {
+        case PLAIN_TEXT:
+            new PlainTextReport(trace, newline)
+                    .renderToStreamWithVerbosityLevel(bufferedOutputStream, verbosity);
+            break;
+        case HTML:
+            new HtmlReport(trace, newline).renderToStreamWithVerbosityLevel(bufferedOutputStream,
+                    verbosity);
+            break;
+        default:
+            throw new IllegalArgumentException(
+                    "Unable to create report with format \"" + format + "\"");
+        }
+        try
+        {
+            bufferedOutputStream.flush();
```

Comment:
I am not sure I am following. The flush is what could cause the IOException here. Do you mean the `renderTo...()` methods should require a BufferedOutputStream as paramter?

Reply:
Why do you need a BufferedOutputStream here? If you really need one: require it as a parameter.

Reply:
My reasoning was that it is up to the code that creates the report to decide which kind of output stream it is going to be. Now that I am thinking about it, it might make more sense to have the `BufferedOutputStream` in an abstract base class of all reports. This way the interface stays clean.

While I am at it I could also exchange the `PrintStream` for a `PrintWriter` since that is probably more efficient in our use case.